docs/src/architecture/assets.md

```diff
@@ -0,0 +1,57 @@
+
+Specifically, the first element will be
+
+   * `ZERO` for a fungible asset: `[e0, ZERO, e2, e3]`
+   * `non-ZERO` for a non-fungible asset: `[e0, non-ZERO, e2, e3]`
+  
+and the most significant bit will be
+
+   * `ONE` for a fungible asset
+   * `ZERO` for a non-fungible asset
+
+The above properties guarantee that there can never be a collision between a fungible and a non-fungible asset. 
+
+## Issuance
+Specialized asset accounts called faucets issue assets by creating notes. Like any other account faucets have an Account ID called `faucet_id`. Users will be able to create faucets on their own. This is similar to creating ERC20 contracts on Ethereum. The methodology for constructing fungible and non-fungible assets is described below.
+
+### Fungible assets
+A fungible asset consists of a `faucet_id` of the faucet which issued the asset as well as the asset amount. The asset amount is guaranteed to be `2^63 - 1` or smaller.
+
+The most significant element of a fungible asset is set to the ID of the faucet which issued
+the asset. This guarantees the properties described above (the first bit is `ONE`).
+
+The least significant element is set to the amount of the asset. This amount cannot be greater
+than `2^63 - 1` and thus requires 63-bits to store. Elements 1 and 2 are set to `ZERO`.
+
+Note: Miden uses little-endian, so a fungible asset looks like `[amount, ZERO, e2, faucet_id]` wheras `e3` as bit ends with `1`
+
+It is impossible to find a collision between two fungible assets issued by different faucets as
+the `faucet_id` is included in the description of the asset and this is guaranteed to be different
+for each faucet as per the faucet creation logic.
+
+### Non-fungible assets
+A non-fungible asset consists of 4 field elements which are computed by hashing asset data (which can be of arbitrary length) to produce: `[e0, e1, e2, e3]`.  We then replace `e1` with the `faucet_id` that issued the asset: `[e0, faucet_id, e2, e3]` to comply with the above rule. We then set the most significant bit to `ZERO`.
+
+Note: Miden uses little-endian, so a non-fungible asset looks like `[e0, faucet_id, e2, e3]` wheras `e3` as bit ends with `0`
+
+It is impossible to find a collision between two non-fungible assets issued by different faucets as the `faucet_id` is included in the description of the non-fungible asset and this is guaranteed to be different as per the faucet creation logic. Collision resistance for non-fungible assets issued by the same faucet is `~2^95`.
+
+## Storage
+Assets are stored in accounts and notes. Both have vaults to record the `asset_id` and the respective `amount`. Accounts can store an unlimited number of assets in a tiered sparse Merkle tree called `asset_tree`. Notes can always only store one type of asset. **ToDO: Can one note store multiple NFTs?**
+
+Depending on the storage mode of the account or the note the information thereof can be private. This is true for any asset in Miden. 
```

Comment:
Notes can store any type of assets but the total number of assets which a note can carry is limited to 255, I believe.

I would also maybe add drawings here to illustrate the difference between account and note vaults.

Reply:
thanks